src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -34,14 +36,19 @@ public ShoppingCartController(ShoppingCartService shoppingCartService,
     }
 
     @PutMapping("/movie-sessions")
-    public void addToCart(@RequestParam Long userId, @RequestParam Long movieSessionId) {
+    public void addToCart(Authentication authentication, @RequestParam Long movieSessionId) {
         shoppingCartService.addSession(
-                movieSessionService.get(movieSessionId), userService.get(userId));
+                movieSessionService.get(movieSessionId),
+                userService.findByEmail(authentication.getName()).orElseThrow(
+                        () -> new RuntimeException("Couldn't add movie session: " + movieSessionId
```

Comment:
In this case, you throw "Couldn't add movie session" if user not found.
So, I consider you have to provide other error message.

Also, you have a duplicated code in this method and in "/by-user" mapping method related to finding user and throwing an exception in case if user not found.

You can do something like:
`private User getUser(Authentication authentication) {
    return userService.findByEmail(authentication.getName())
            .orElseThrow(() -> new NoSuchElementException("User not found: " + authentication.getName()));
}`